### PR TITLE
Fixed config type cast

### PR DESCRIPTION
**Pull request by abi87**
## Why this should be merged
A faulty type cast that won't log correctly. This PR fixes it.

## How this works
Just log the right structs

## How this was tested
CI

### CODE REVIEW

File: x/warp/module.go [hunk 1]
```diff
@@ -48,9 +48,8 @@ func (*configurator) MakeConfig() precompileconfig.Config {
 
 // Configure is a no-op for warp since it does not need to store any information in the state
 func (*configurator) Configure(chainConfig precompileconfig.ChainConfig, cfg precompileconfig.Config, state contract.StateDB, _ contract.ConfigurationBlockContext) error {
-	config, ok := cfg.(*Config)
-	if !ok {
-		return fmt.Errorf("incorrect config %T: %v", config, config)
+	if _, ok := cfg.(*Config); !ok {
+		return fmt.Errorf("expected config type %T, got %T: %v", &Config{}, cfg, cfg)
```
Comment (ceyonur) [inline x/warp/module.go:52]: We do have same issue in other precompiles + template. Could you do a search + replace to fix others as well?

**Review by ceyonur — CHANGES_REQUESTED** [review]
We should fix other places as well.

File: x/warp/module.go (hunk 1 above)
Comment (abi87) [reply x/warp/module.go:52]: Done.

**Review by ceyonur — APPROVED** [review]

File: precompile/contracts/txallowlist/module.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func (*configurator) MakeConfig() precompileconfig.Config {
 func (*configurator) Configure(chainConfig precompileconfig.ChainConfig, cfg precompileconfig.Config, state contract.StateDB, blockContext contract.ConfigurationBlockContext) error {
 	config, ok := cfg.(*Config)
```
Comment (ceyonur) [inline precompile/contracts/txallowlist/module.go:44]: hmm we should probably replace `config` with `_` as we don't use it (to fix UTs)

**Review by darioush — APPROVED** [review]

**Pull request merged**